### DISCUSSION TITLE
Anyone else having a bunch of issues past couple of days?

### DISCUSSION THREAD

**Discussion by treeder** [community/community#10129] [Codespaces]
Issues like: 

* Taking forever to load extensions, if they load at all
* No "ports" tab shows up
* Git tab shows as it not being initialized
* Extremely slow terminal response
* Extremely slow loading files

**Comment by treeder** [comment]
I see a bunch of disconnecting and reconnecting messages in the "Window" logs, like this:

![image](https://user-images.githubusercontent.com/75826/149839855-dff59e48-dbe5-46e9-822f-2ec860513a4f.png)

Sometimes it boots me out of the codespace and says the limit of 30 guests was reached.  I didn't capture the full message before restarting, but will try next time. 

Past week has been brutal for codespaces...

**Comment by treeder** [comment]
![image](https://user-images.githubusercontent.com/75826/149840379-717e1565-e354-451a-ad4d-97d702f54758.png)

**Comment by treeder** [comment]
![image](https://user-images.githubusercontent.com/75826/149840549-33ea11bb-0ca9-4c00-b77f-4db0d5dc372a.png)

**Comment by bradtaniguchi** [comment]
I'm also seeing issues with my personal codespaces on multiple repo's. I've re-built them and see similar problems as you have.

No idea how to fix it, or what is going on. But at the moment codespaces is basically unusable.

**update**

I also wanted to give out, I'm using a Pixelbook i7 as where I use codespaces via a PWA install. Out of curiosity I opened up an existing codespace from my windows desktop and I'm not seeing any problems.

I'll try this out on other machines to see if maybe its related to my actual device at some level.


**update**
System specs:
Chrome Version+Chrome OS: 98.0.4758.51 (Official Build) beta (64-bit)

My other system (Windows 11), which has codespaces running fine, is Chrome Version 97.0.4692.71.

I'll try out codespaces on a beta version on the same system, where things are working to see if the issue is tied to the chrome version. Maybe some change with chrome is affected the performance of codespaces?

**Reply by treeder** [reply]
Yes, it is unusable. Wasted so much time trying to get this working properly... Guess it's time to give up and go back to local.

**Comment by bradtaniguchi** [comment]
I just installed chrome beta onto my windows 11 machine (which had codespaces running fine with chrome 97).

**And 98 (beta) ran into the same issues as what I see on my chromebook on the same chrome version**.

I'd be left to assume something in chrome 98 (beta) is what is causing the issues. 

@treeder What browser version and OS were you running codespaces on?

**Reply by treeder** [reply]
Ahh, that's possible. I am on Chrome OS 98 too.

**Reply by treeder** [reply]
Possibly this: https://bugs.chromium.org/p/chromium/issues/detail?id=1286909&sort=-opened%20-modified&q=FoundIn%3D98&can=2

**Reply by treeder** [reply]
Don't think so, my upload speed on a couple of speed test sites is fine. The only issue I'm having is on Codespaces.

**Reply by bradtaniguchi** [reply]
I did more digging with a "dual install" so I could inspect the codespace as it runs in the browser compared to a working solution. 

I noticed the "slow" codespace gets there error:
![image](https://user-images.githubusercontent.com/10079147/150274611-f5a420c0-6848-4445-995a-7c6a8059bc07.png)
(might be some sensitive info related to the "web-client", but at this rate I might not ever be able to use codespaces once the new chrome version drops lol)

So for whatever reason, the `define` function doesn't exist within the page in chrome 98.

Along with that initial error, there will be more continual errors that all end with:
```
Ssh channel failed to open, retry: 2. Error: Opening SSH channel failed, Timed out.
```
It appears all proceeding performance issues are tied with the `WebSocketFactory.js` struggling to get a connection. For whatever reason the socket seems to get closed almost immediately, and thus the editor becomes nearly useless as the socket tries to re-establish connection, gets a connection after re-trying+backoff, then it gets disposed again, hence why things seem "slow" even though there seems to be bursts of stability.

I'm not sure if github-support would even cover this case, since I'm using a beta browser on an individual plan/early-access. But I might try to reach out to them directly. Since it doesn't seem like we can really do anything.

**Reply by treeder** [reply]
I rolled back to Chrome OS stable 97.0.4692.91 and lo and behold, it works. Hopefully Chrome and GitHub folks figure this out before next stable release or there's going to be a lot of unhappy Codespaces users.

**Reply by bradtaniguchi** [reply]
I'm going to open a ticket up with github support, at least to let them know that this could create more issues once chrome 98 drops. 

I'll link this discussion in the ticket, at least to give some weight it isn't a one off.

I'll report back if I hear anything, in the mean time I'll stay on 98 for now. :)

**Comment by jkeech** [comment]
👋 Hi from the Codespaces team. Thanks for raising this issue! We are investigating this issue on Chrome 98, and it does appear to be related to a Chrome WebSocket bug which is being fixed before Chrome 98 ships to Stable: https://bugs.chromium.org/p/chromium/issues/detail?id=1286909#c12. It sounds like the fix will get merged into Chrome 98 early next week.

**Reply by jkeech** [reply]
Chrome released a new build to the Beta channel today (`Version 98.0.4758.74 (Official Build) beta (arm64)`) which contains the fix. Codespace connections are stable on the new beta build. https://chromereleases.googleblog.com/2022/01/beta-channel-update-for-desktop_27.html

Please let us know if you continue to see issues after updating to the new build. Thanks!

**Reply by bradtaniguchi** [reply]
I just did a quick check with my windows machine on chrome beta (with that version above) and everything was working perfectly. 

Unfortunately my beta-channel ChromeOS still sees issues even after the Tuesday update (98.0.4758.67). I'd assume this will be fixed sometime next week with the next release for ChromeOS beta channel.

**Reply by bradtaniguchi** [reply]
With the latest beta channel update in chromeOS, of Google Chrome: `98.0.4758.79` the issue seems to be resolved, and codespaces is back to working quick as ever. :D

**Reply by jkeech** [reply]
Thanks for confirming! Glad to hear it's working well again 🎉.
